packages/client/hmi-client/src/page/data-explorer/components/tera-search-results-list.vue

```diff
@@ -99,9 +101,62 @@ const props = defineProps({
 	}
 });
 
-const previewedAsset = ref<ResultType | null>(null);
+const selectedAsset = ref();
 
-const emit = defineEmits(['toggle-data-item-selected']);
+const projectOptions = computed(() => [
+	{
+		label: 'Add this to...',
+		items:
+			useProjects().allProjects.value?.map((project) => ({
+				label: project.name,
+				command: async () => {
+					let response: any = null;
+					let assetName = '';
+
+					if (isDocument(selectedAsset.value)) {
+						const document = selectedAsset.value as Document;
+						await createDocumentFromXDD(document, project.id);
+						// finally add asset to project
+						response = await useProjects().get(project.id);
+						assetName = selectedAsset.value.title;
+					}
+					if (isModel(selectedAsset.value)) {
+						// FIXME: handle cases where assets is already added to the project
+						const modelId = selectedAsset.value.id;
+						// then, link and store in the project assets
+						const assetType = AssetType.Models;
+						response = await useProjects().addAsset(assetType, modelId, project.id);
+						assetName = selectedAsset.value.header.name;
+					}
+					if (isDataset(selectedAsset.value)) {
+						// FIXME: handle cases where assets is already added to the project
+						const datasetId = selectedAsset.value.id;
+						// then, link and store in the project assets
+						const assetType = AssetType.Datasets;
+						if (datasetId) {
+							response = await useProjects().addAsset(assetType, datasetId, project.id);
+							assetName = selectedAsset.value.name;
+						}
+					}
+					console.log(selectedAsset.value);
+
+					if (response) logger.info(`Added ${assetName} to ${project.name}`);
```

Comment:
Add an error handling with a `logger.warn` explaining that the asset was not added to the project.